ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/TransactionPoolFactory.java

```diff
@@ -35,13 +37,16 @@ public static TransactionPool createTransactionPool(
     final PeerTransactionTracker transactionTracker = new PeerTransactionTracker();
     final TransactionsMessageSender transactionsMessageSender =
         new TransactionsMessageSender(transactionTracker);
+    final MutableBlockchain blockchain = protocolContext.getBlockchain();
+    final SyncState syncState = new SyncState(blockchain, ethContext.getEthPeers());
```

Comment:
We should be using the shared SyncState that the controllers use.  So add this as a param to `createTransactionPool` and have the callers pass their instance.  They all should have one by the point they make this call.

Reply:
yeah -- that makes sense. branch has been updated accordingly